### PR TITLE
Meson Starter Project

### PR DESCRIPTION
**Pull request by slci**
- sample project that uses Meson and Conan
- testing framework that integrates gmock and Catch2

### CODE REVIEW

File: meson_starter_project/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+```sh
+# Install conan & meson
+pip3 install --user --upgrade conan meson
+
+# Create build directory
+mkdir build && cd build
+
+# Add build dir to pkg-config search path
+export PKG_CONFIG_PATH=$(pwd)
```
Comment (fawcio) [inline meson_starter_project/README.md:15]: This is ugly! I don't know how to get rid of this export.
I think it's a limitation of Meson. I haven't found a way to make Meson set it automatically.

File: meson_starter_project/conan/meson.build [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Run Conan dependency installation as external command
+
+conan = find_program('conan')
+message('Running Conan dependency installation...')
+# conan_out = run_command(conan, 'install', '--build=missing', '-if', '$MESON_BUILD_ROOT', '$MESON_SOURCE_ROOT/conan', check: true)
+
+# Workaround for Meson not substituting $MESON_BUILD_ROOT and $MESON_SOURCE_ROOT in direct call of conan.
+env = find_program('env')
+conan_out = run_command(env, 'sh', '-c',
```
Comment (fawcio) [inline meson_starter_project/conan/meson.build:9]: This is shitty as well. I'm wondering about rising an issue for Meson about that. Does anyone know what could be wrong here?

File: meson_starter_project/README.md (hunk 1 above)
Comment (bburda) [reply meson_starter_project/README.md:15]: Well, let's just leave it for now. We will change it as soon as we find a better solution.

File: meson_starter_project/conan/meson.build (hunk 2 above)
Comment (bburda) [reply meson_starter_project/conan/meson.build:9]: Same as above, let's leave it for now. You can raise an issue for Meson and change it when you find a better alternative.

**Pull request merged**